hack/e2e/run-rp-and-e2e.sh

```diff
@@ -1,6 +1,16 @@
 #!/bin/bash -e
 ######## Helper file to run E2e either locally or using Azure DevOps Pipelines ########
 
+# set environment required for each step of e2e
+set -o pipefail
+set -x
+az account set -s $AZURE_SUBSCRIPTION_ID
+export SECRET_SA_ACCOUNT_NAME=$(SECRET_SA_ACCOUNT_NAME)
+export CI=true
+make secrets
+. secrets/env
+echo "##vso[task.setvariable variable=RP_MODE]$RP_MODE"
```

Comment:
This script is also used when running e2e tests locally. Aside from having ADO pipeline-specific functionality being echoed (ultimately harmless), will re-running `make secrets` or setting variables such as `CI=true` break local testing?

Reply:
Which script used for local testing are you referring to? I typically wouldn't try and source `hack/e2e/run-rp-and-e2e.sh` when testing locally - I'd just be running `make test-e2e` with an admin kubeconfig.

When you say re-running `make secrets`, do you mean running it at the beginning of each step in the pipeline?

Reply:
The [documentation for running e2e tests locally](https://github.com/Azure/ARO-RP/blob/master/docs/testing.md#e2e-tests) has the following:

> * Source the [helper script](https://github.com/Azure/ARO-RP/blob/master/hack/e2e/run-rp-and-e2e.sh) to set the proper ENV variables. If you run the tests locally, run export LOCAL_E2E=true env before sourcing the helper file.

Additionally, the second line of this file has this comment:

> `######## Helper file to run E2e either locally or using Azure DevOps Pipelines ########`

This documentation is what I followed for my recent stories running e2e tests against a locally-running RP. If that process has changed then the documentation needs to be updated.

> When you say re-running `make secrets`, do you mean running it at the beginning of each step in the pipeline?

No, I mean locally. Running `make secrets` will fail unless `SECRET_SA_ACCOUNT_NAME` is set, and since the [dev RP deploy instructions](https://github.com/Azure/ARO-RP/blob/master/docs/deploy-development-rp.md#prepare-your-environment) have it set in-line and the environment file that secret gets saved to is `secrets/env` which is not sourced until the next line, it is not guaranteed that someone will have this in their environment variables prior to running this script.

Reply:
I wouldn't mind just changing the doc, it's easier than making `hack/e2e/run-rp-and-e2e.sh` function for both workflows. What do you think?

Reply:
Hmm, on second look I think I see a path forward for keeping both workflows - I'll continue on this wednesday and see what I can do.

Reply:
After giving it some thought I think splitting the workflows up may be the better solution. I'm a big fan of Do One Thing, and the way this script is evolving is looking more and more like it's trying to do multiple different things and suffering for it. I think scripts for running things in CI and scripts for running things locally should probably be dealt with separately if the functionality starts diverging. Personally I'd also keep the CI scripts in the `.pipelines` folder so the use case for them is clear.

Reply:
We chatted about this a bit, but since [LOCAL_E2E testing method](https://github.com/Azure/ARO-RP/blob/master/docs/testing.md#e2e-tests) is still in use and we would have to source `hack/e2e/run-rp-and-e2e.sh` for its functions to keep things DRY, I'm going to keep it all in `run-rp-and-e2e.sh` for this PR rather than have the `CI=true` code in its own script in `.pipelines`. This approach could definitely be revisited in a future PR if needed.